lib/elixir/lib/kernel.ex

```diff
@@ -3210,6 +3210,10 @@ defmodule Kernel do
   """
   defmacro defstruct(fields) do
     quote bind_quoted: [fields: fields] do
+      if Module.get_attribute(__MODULE__, :struct) do
+        raise ArgumentError, "defstruct has already been called for " <>
+          "#{Macro.to_string(__MODULE__)}, defstruct can only be called once per module"
```

Comment:
Indeed, it has to be `inspect/1`, `Macro.to_string/1` will miss all the clauses and "delegate" to `inspect/1` call.